src/main/java/com/minecolonies/coremod/colony/buildings/workerbuildings/BuildingCook.java

```diff
@@ -115,8 +115,7 @@
     public BuildingCook(final IColony c, final BlockPos l)
     {
         super(c, l);
-        keepX.put(ItemStackUtils.ISFOOD, new Tuple<>(STACKSIZE, true));
-        keepX.put(ItemStackUtils.ISCOOKABLE, new Tuple<>(STACKSIZE, true));
+        keepX.put(stack -> isAllowedItem("food", new ItemStorage(stack)), new Tuple<> (STACKSIZE, true));
         keepX.put(stack -> isAllowedFuel(stack), new Tuple<>(STACKSIZE, true));
```

Comment:
@someaddons is right, if you used the allowedItems as "blacklist" similar to the lumberjack. Those should be inverted.